### PR TITLE
caf 0.18.6 zeek 5.0.1

### PR DESCRIPTION
**Pull request by chenrui333**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by danielnachun** [comment]
`zeek` needs a revision bump.

**Comment by carlocab** [comment]
```
  In file included from /tmp/zeek-20220911-13615-1mtbn59/auxil/broker/include/broker/atoms.hh:5:
  /tmp/zeek-20220911-13615-1mtbn59/auxil/broker/include/broker/fwd.hh:142:20: error: no template named 'has_type_id' in namespace 'caf'; did you mean 'has_type_id_v'?
  static_assert(caf::has_type_id<broker::timespan>::value,
                ~~~~~^~~~~~~~~~~
                     has_type_id_v
  /opt/homebrew/opt/caf/include/caf/type_id.hpp:79:16: note: 'has_type_id_v' declared here
  constexpr bool has_type_id_v = detail::is_complete<type_id<T>>;
                 ^
```

**Comment by danielnachun** [comment]
We should try to update `zeek` to 5.0.1 as we are currently on 4.2.2 and this older version seems to have an API incompatibility with `caf` 0.18.6.

File: Formula/zeek.rb [hunk 1]
```diff
@@ -2,9 +2,10 @@ class Zeek < Formula
   desc "Network security monitor"
   homepage "https://www.zeek.org"
   url "https://github.com/zeek/zeek.git",
-      tag:      "v4.2.2",
-      revision: "40eb7f80378284202e52e6c45299cac10abf07ab"
+      tag:      "v5.0.1",
+      revision: "66eaf5d7c769ee65588e4b14b39203a55f39b3ac"
   license "BSD-3-Clause"
+  revision 1
```
Comment (carlocab) [inline Formula/zeek.rb:8]:
```suggestion
```
Also, needs `flex`.

**Comment by carlocab** [comment]
Cherry-picked onto #111218.

**Comment by carlocab** [comment]
Nope, `zeek` still doesn't like this version of `caf`. This will likely need a rebase.

**Pull request closed without merging**